src/widgets/InfoBadge.js

```diff
@@ -1,129 +1,103 @@
+  // Get total of all the count variables in the info array. We want to show it on the badge
+  const info = getBadgeData(routeName);
+  const pendingCount = info.reduce((total, item) => total + (item.count || 0), 0);
 
-    // Get total of all the count variables in the info array. We want to show it on the badge
-    const pendingCount = info.reduce((total, item) => total + (item.count || 0), 0);
+  // show 99+ if the number is greater then 99 to limit the number of characters.
+  const unfinalisedCountText = pendingCount > 99 ? '99+' : pendingCount;
 
-    // Since having multiple digit text can break the badge, we are showing 99+
-    // if the number is greater then 99.
-    const unfinalisedCountText = pendingCount > 99 ? '99+' : pendingCount;
+  const popoverText = useCallback(
+    (item, key) =>
+      (item.count > 0 || item.text) && (
+        <Text style={badgeTextStyle} key={key}>
+          {`${item.title}: ${item.count || item.text}`}
+        </Text>
+      ),
+    []
+  );
 
-    return (
-      <View style={{ position: 'relative' }}>
-        {children}
-        {pendingCount !== 0 && (
-          <View style={[localStyles.mainWrapperStyle, mainWrapperStyle]}>
-            <TouchableHighlight
-              ref={ref => {
-                this.touchableContainer = ref;
-              }}
-              style={localStyles.touchableContainerStyle}
-              onPress={() => this.showPopover()}
-            >
-              <Badge
-                value={unfinalisedCountText}
-                badgeStyle={localStyles.badgeStyle}
-                textStyle={localStyles.badgeFontStyle}
-              />
-            </TouchableHighlight>
-            <Popover
-              isVisible={isPopOverVisible}
-              fromRect={rect}
-              onRequestClose={() => this.closePopover()}
-              popoverStyle={{ padding: 10, backgroundColor: SUSSOL_ORANGE }}
-              arrowStyle={{ backgroundColor: SUSSOL_ORANGE }}
-              backgroundStyle={{ backgroundColor: 'transparent' }}
-              placement={popoverPosition}
-            >
-              <Text style={localStyles.badgeTextStyle}>
-                {info.map(
-                  (item, key) =>
-                    (item.count > 0 || item.text) && (
-                      // eslint-disable-next-line react/no-array-index-key
-                      <Text key={key}>
-                        {item.title} : {item.count || item.text}
-                        {key === info.length - 1 ? '' : '\n'}
-                      </Text>
-                    )
-                )}
-              </Text>
-            </Popover>
-          </View>
-        )}
-      </View>
-    );
-  }
-}
+  return (
+    <View>
+      {children}
+      {pendingCount && (
+        <View style={mainWrapperStyle}>
+          <TouchableHighlight ref={ref} style={touchableContainerStyle} onPress={show}>
+            <Badge value={unfinalisedCountText} />
+          </TouchableHighlight>
```

Comment:
@joshxg I think I have a solution for the error above. 

The conditional above excepts the left side expression to be a boolean, simply checking `pendingCount` would give this error. 

```
pendingCount !== 0
```
...fixes this.

Reply:
oops! 4a7447f81dbdf2e2c73011b10e6f036ee3ce7e33